### PR TITLE
Support more configuration options

### LINKED ISSUE #7: Support more config options
Thanks for this useful shim!

I'm getting some warnings because there isn't a way to set `--no-toc`, and I'd also like to be able to set `--module-first`. It would be great to have a way to provide those options.

### PR DESCRIPTION
**Pull request by paulmelnikow**
Closes #7 

I wasn't sure how to test this, beyond local manual testing. Let me know if there's something simple I can do there. Thanks!

### CODE REVIEW

**Comment by stephenfin** [comment]
This all looks good. Thanks for stepping up here. One request...

> I wasn't sure how to test this, beyond local manual testing. Let me know if there's something simple I can do there. Thanks!

There are basic tests in `tests/test_ext.py`. Is this something you could extend? It should be possible to test the `apidoc_toc_file` and `apidoc_module_first` options by checking for the presence of your chosen ToC filename and the ordering of lines in the ToC file, respectively. I'm not sure how to test the `apidoc_extra_args`, though I suppose you could simply enable one of the other two options manually through this option? Give me a shout if you need any help.

**Comment by paulmelnikow** [comment]
Okay, I'll give it a shot.

Where does this assert exists functionality come from?

```py
assert (app.outdir / 'api' / 'apidoc_dummy_package.html').exists()
```

Is there a way to use it to snag the contents of a file?

**Comment by stephenfin** [comment]
> Okay, I'll give it a shot.

I saw this too late :smile: I've gone and implemented those tests. Sorry if you'd invested time in this :see_no_evil: Fancy reviewing those patches though to see if anything is different from what you'd expect?

> Where does this "assert exists" functionality come from?
> 
> ```python
> assert (app.outdir / 'api' / 'apidoc_dummy_package.html').exists()
> ```
> 
> Is there a way to use it to snag the contents of a file?

That's pytest magicness and I think it's awful. `app.outdir` here is an instance of [`sphinx.testing.path.path`](https://github.com/sphinx-doc/sphinx/blob/323d27ef8f63b8bb2e351ddf5fd2cd89abb97df8/sphinx/testing/path.py). That class overrides the '__div__' operator to mean concatenation which is...confusing, to say the least. While we're at it, `app` itself, which is defined [here](https://github.com/sphinx-doc/sphinx/blob/323d27ef8f63b8bb2e351ddf5fd2cd89abb97df8/sphinx/testing/fixtures.py#L113) is an example of a pytest fixture, which is enabled [using function arguments](https://docs.pytest.org/en/latest/fixture.html) rather than explicitly called. Again, ick

\</rant>

File: sphinxcontrib/apidoc/__init__.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def setup(app):
     app.add_config_value('apidoc_output_dir', 'api', 'env', [str])
     app.add_config_value('apidoc_excluded_paths', [], 'env', [[str]])
     app.add_config_value('apidoc_separate_modules', False, 'env', [bool])
-    app.add_config_value('apidoc_toc_file', 'modules', 'env', [str, bool])
+    app.add_config_value('apidoc_toc_file', None, 'env', [str, bool])
```
Comment (paulmelnikow) [inline sphinxcontrib/apidoc/__init__.py]: 👍

**Comment by paulmelnikow** [comment]
While I really appreciate your work on the library, I feel like we miscommunicated there. You asked me to do it, and so I did, but then you did it yourself…

Nevertheless, thanks for wrapping this up. You may want to adapt what I did in https://github.com/paulmelnikow/apidoc/commit/92e2a18c4a42aafb3f83dde96f6cf5dd432fa3bb, to test `apidoc_extra_args` using `--private`.

**Comment by paulmelnikow** [comment]
It's all good. Yea, please go for it!

**Comment by paulmelnikow** [comment]
Thanks so much!

Could you do a release?

**Comment by stephenfin** [comment]
Yup, done. Travis should have done this for me but, alas, no luck

**Pull request merged**